src/Sylius/Bundle/UserBundle/Resources/config/services.xml

```diff
@@ -24,7 +24,10 @@
         <parameter key="sylius.user.canonicalizer.class">Sylius\Component\User\Canonicalizer\Canonicalizer</parameter>
         <parameter key="sylius.user.password_encoder.class">Sylius\Bundle\UserBundle\Security\UserPasswordEncoder</parameter>
         <parameter key="sylius.user.password_updater.class">Sylius\Component\User\Security\PasswordUpdater</parameter>
-        <parameter key="sylius.user.code_provider.class">Sylius\Component\User\Security\TokenProvider</parameter>
+
+        <parameter key="sylius.user.token_provider.abstract_unique.class">Sylius\Component\User\Security\Provider\AbstractUniqueTokenProvider</parameter>
+        <parameter key="sylius.user.token_provider.password_reset.class">Sylius\Component\User\Security\Provider\PasswordResetTokenProvider</parameter>
```

Comment:
What is the difference in providing/generating a unique token? A logic is same, only one thing that change is where we check token uniqueness. Maybe we should extract a Token generation logic, and just inject a unique checker?